src/olympia/blocklist/utils.py

```diff
@@ -2,6 +2,17 @@
 from olympia.activity import log_create
 
 
+def add_to_latest_version_in_channel(obj, al, channel):
+    from olympia.activity.models import VersionLog
+
+    version_ids = [
+        id_ for id_, chan in obj.addon_versions.values()
```

Comment:
I don't think it makes sense to use `addon_versions` here: it can be super costly (both in python and db), and unless I'm mistaken it doesn't appear to be used by anything else when saving a block, so you don't gain anything by preloading it. In this case you just need the latest version in each channel, it's probably best to use `find_latest_version()` instead, even if that means making 2 queries.

Reply:
For a single block `addon_versions` will have been preloaded for `_get_version_choices`, called by `get_form`.
For multiple blocks, looks like you're right - though it should be more efficient to add in a line to specifically call `Block.preload_addon_versions` and do one (big) query for all Blocks being saved than 2 small queries per Block. (?)

Reply:
Ah, I hadn't thought about the fact that the choices would have been generated for the single block, makes sense.

For multiple blocks... preloading *all* versions from *all* addons in a multiple block just to access 2 of them per add-on seems a bit much... and likely to cause issues depending on how many add-ons are in the block, since the block is processed by a single task.

Reply:
Well, we're going to need all the versions to be able to add a `VersionLog` entry for all of them anyway.  

I was thinking the submit task might need changing to batch the guids tbh.  Running `process_input_guids` on a slice of 100 guids each time.  (I'd not planned to add it in this PR)

Reply:
Yeah. It's fine to do in a follow-up, I guess. I'm slightly worried about the number of versions we can end up loading and then inserting a `VersionLog` for, but I guess we already have similar problems when we display a single block form anyway (albeit without the insertion part)